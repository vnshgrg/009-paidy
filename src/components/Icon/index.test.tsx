import "react-native";
import React from "react";
import { Icon } from "./index";

import renderer from "react-test-renderer";

it("renders CheckIcon correctly", () => {
  const tree = renderer.create(<Icon icon="check"></Icon>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders ListBulletIcon correctly", () => {
  const tree = renderer.create(<Icon icon="list-bullet"></Icon>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders LockIcon correctly", () => {
  const tree = renderer.create(<Icon icon="lock"></Icon>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders UnlockIcon correctly", () => {
  const tree = renderer.create(<Icon icon="unlock"></Icon>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders MinusIcon correctly", () => {
  const tree = renderer.create(<Icon icon="minus"></Icon>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders PlusIcon correctly", () => {
  const tree = renderer.create(<Icon icon="plus"></Icon>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders PlusBicIcon correctly", () => {
  const tree = renderer.create(<Icon icon="plus-big"></Icon>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders TrashIcon correctly", () => {
  const tree = renderer.create(<Icon icon="trash"></Icon>).toJSON();
  expect(tree).toMatchSnapshot();
});
