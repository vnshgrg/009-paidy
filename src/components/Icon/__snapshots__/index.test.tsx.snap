// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CheckIcon correctly 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight="24"
  bbWidth="24"
  fill="none"
  focusable={false}
  height={24}
  meetOrSlice={0}
  minX={0}
  minY={0}
  stroke="#0288D1"
  strokeWidth={2}
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "flex": 0,
        "height": 24,
        "width": 24,
      },
    ]
  }
  vbHeight={24}
  vbWidth={24}
  width={24}
>
  <RNSVGGroup
    fill={null}
    propList={
      [
        "fill",
        "stroke",
        "strokeWidth",
      ]
    }
    stroke={
      {
        "payload": 4278356177,
        "type": 0,
      }
    }
    strokeWidth="2"
  >
    <RNSVGPath
      d="M9 12.75L11.25 15 15 9.75M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;

exports[`renders ListBulletIcon correctly 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight="24"
  bbWidth="24"
  fill="none"
  focusable={false}
  height={24}
  meetOrSlice={0}
  minX={0}
  minY={0}
  stroke="#0288D1"
  strokeWidth={2}
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "flex": 0,
        "height": 24,
        "width": 24,
      },
    ]
  }
  vbHeight={24}
  vbWidth={24}
  width={24}
>
  <RNSVGGroup
    fill={null}
    propList={
      [
        "fill",
        "stroke",
        "strokeWidth",
      ]
    }
    stroke={
      {
        "payload": 4278356177,
        "type": 0,
      }
    }
    strokeWidth="2"
  >
    <RNSVGPath
      d="M8.25 6.75h12M8.25 12h12m-12 5.25h12M3.75 6.75h.007v.008H3.75V6.75zm.375 0a.375.375 0 11-.75 0 .375.375 0 01.75 0zM3.75 12h.007v.008H3.75V12zm.375 0a.375.375 0 11-.75 0 .375.375 0 01.75 0zm-.375 5.25h.007v.008H3.75v-.008zm.375 0a.375.375 0 11-.75 0 .375.375 0 01.75 0z"
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;

exports[`renders LockIcon correctly 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight="24"
  bbWidth="24"
  fill="none"
  focusable={false}
  height={24}
  meetOrSlice={0}
  minX={0}
  minY={0}
  stroke="#0288D1"
  strokeWidth={2}
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "flex": 0,
        "height": 24,
        "width": 24,
      },
    ]
  }
  vbHeight={24}
  vbWidth={24}
  width={24}
>
  <RNSVGGroup
    fill={null}
    propList={
      [
        "fill",
        "stroke",
        "strokeWidth",
      ]
    }
    stroke={
      {
        "payload": 4278356177,
        "type": 0,
      }
    }
    strokeWidth="2"
  >
    <RNSVGPath
      d="M16.5 10.5V6.75a4.5 4.5 0 10-9 0v3.75m-.75 11.25h10.5a2.25 2.25 0 002.25-2.25v-6.75a2.25 2.25 0 00-2.25-2.25H6.75a2.25 2.25 0 00-2.25 2.25v6.75a2.25 2.25 0 002.25 2.25z"
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;

exports[`renders MinusIcon correctly 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight="24"
  bbWidth="24"
  fill="none"
  focusable={false}
  height={24}
  meetOrSlice={0}
  minX={0}
  minY={0}
  stroke="#0288D1"
  strokeWidth={2}
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "flex": 0,
        "height": 24,
        "width": 24,
      },
    ]
  }
  vbHeight={24}
  vbWidth={24}
  width={24}
>
  <RNSVGGroup
    fill={null}
    propList={
      [
        "fill",
        "stroke",
        "strokeWidth",
      ]
    }
    stroke={
      {
        "payload": 4278356177,
        "type": 0,
      }
    }
    strokeWidth="2"
  >
    <RNSVGPath
      d="M15 12H9m12 0a9 9 0 11-18 0 9 9 0 0118 0z"
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;

exports[`renders PlusBicIcon correctly 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight="24"
  bbWidth="24"
  fill="none"
  focusable={false}
  height={24}
  meetOrSlice={0}
  minX={0}
  minY={0}
  stroke="#0288D1"
  strokeWidth={2}
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "flex": 0,
        "height": 24,
        "width": 24,
      },
    ]
  }
  vbHeight={24}
  vbWidth={24}
  width={24}
>
  <RNSVGGroup
    fill={null}
    propList={
      [
        "fill",
        "stroke",
        "strokeWidth",
      ]
    }
    stroke={
      {
        "payload": 4278356177,
        "type": 0,
      }
    }
    strokeWidth="2"
  >
    <RNSVGPath
      d="M12 4.5v15m7.5-7.5h-15"
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;

exports[`renders PlusIcon correctly 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight="24"
  bbWidth="24"
  fill="none"
  focusable={false}
  height={24}
  meetOrSlice={0}
  minX={0}
  minY={0}
  stroke="#0288D1"
  strokeWidth={2}
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "flex": 0,
        "height": 24,
        "width": 24,
      },
    ]
  }
  vbHeight={24}
  vbWidth={24}
  width={24}
>
  <RNSVGGroup
    fill={null}
    propList={
      [
        "fill",
        "stroke",
        "strokeWidth",
      ]
    }
    stroke={
      {
        "payload": 4278356177,
        "type": 0,
      }
    }
    strokeWidth="2"
  >
    <RNSVGPath
      d="M12 9v6m3-3H9m12 0a9 9 0 11-18 0 9 9 0 0118 0z"
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;

exports[`renders TrashIcon correctly 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight="24"
  bbWidth="24"
  fill="none"
  focusable={false}
  height={24}
  meetOrSlice={0}
  minX={0}
  minY={0}
  stroke="#0288D1"
  strokeWidth={2}
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "flex": 0,
        "height": 24,
        "width": 24,
      },
    ]
  }
  vbHeight={24}
  vbWidth={24}
  width={24}
>
  <RNSVGGroup
    fill={null}
    propList={
      [
        "fill",
        "stroke",
        "strokeWidth",
      ]
    }
    stroke={
      {
        "payload": 4278356177,
        "type": 0,
      }
    }
    strokeWidth="2"
  >
    <RNSVGPath
      d="M14.74 9l-.346 9m-4.788 0L9.26 9m9.968-3.21c.342.052.682.107 1.022.166m-1.022-.165L18.16 19.673a2.25 2.25 0 01-2.244 2.077H8.084a2.25 2.25 0 01-2.244-2.077L4.772 5.79m14.456 0a48.108 48.108 0 00-3.478-.397m-12 .562c.34-.059.68-.114 1.022-.165m0 0a48.11 48.11 0 013.478-.397m7.5 0v-.916c0-1.18-.91-2.164-2.09-2.201a51.964 51.964 0 00-3.32 0c-1.18.037-2.09 1.022-2.09 2.201v.916m7.5 0a48.667 48.667 0 00-7.5 0"
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;

exports[`renders UnlockIcon correctly 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight="24"
  bbWidth="24"
  fill="none"
  focusable={false}
  height={24}
  meetOrSlice={0}
  minX={0}
  minY={0}
  stroke="#0288D1"
  strokeWidth={2}
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "flex": 0,
        "height": 24,
        "width": 24,
      },
    ]
  }
  vbHeight={24}
  vbWidth={24}
  width={24}
>
  <RNSVGGroup
    fill={null}
    propList={
      [
        "fill",
        "stroke",
        "strokeWidth",
      ]
    }
    stroke={
      {
        "payload": 4278356177,
        "type": 0,
      }
    }
    strokeWidth="2"
  >
    <RNSVGPath
      d="M13.5 10.5V6.75a4.5 4.5 0 119 0v3.75M3.75 21.75h10.5a2.25 2.25 0 002.25-2.25v-6.75a2.25 2.25 0 00-2.25-2.25H3.75a2.25 2.25 0 00-2.25 2.25v6.75a2.25 2.25 0 002.25 2.25z"
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;
